benchmarks/persistence/persistence.py

```diff
@@ -0,0 +1,205 @@
+            self.commit_hash,
+        ]
+
+        sanitized_components = [
+            self._string_sanitizer(component) for component in components
+        ]
+        path = separator.join(sanitized_components) + separator + "result.json"
+
+        return path
+
+
+class PersistenceHandlingInterface(Protocol):
+    """Interface for interacting with storage."""
+
+    def write_object(self, path: PathConstructor, object: dict) -> None:
+        """Store a JSON serializable dict according to the path.
+
+        Args:
+            path: The path of the object to be persisted
+            object: The object to be persisted.
+        """
+
+
+@define
+class S3PersistenceHandler:
+    """Class for persisting objects in an S3 bucket."""
+
+    _bucket_name: str = field(validator=instance_of(str), init=False)
+    """The name of the S3 bucket where the results are stored."""
+
+    _object_session = boto3.session.Session()
+    """The boto3 session object. This will load the respective credentials
+    from the environment variables within the container."""
+
+    @_bucket_name.default
+    def _default_bucket_name(self) -> str:
+        """Get the bucket name from the environment variables."""
+        if "BAYBE_PERFORMANCE_PERSISTENCE_PATH" not in os.environ:
+            raise ValueError("No S3 bucket name provided.")
+        return os.environ["BAYBE_PERFORMANCE_PERSISTENCE_PATH"]
+
+    def write_object(self, path: PathConstructor, object: dict) -> None:
+        """Store a JSON serializable dict according to the path.
+
+        This method will store an JSON serializable dict in an S3 bucket.
+        The S3-key of the Java Script Notation Object will be created by
+        the path object.
+
+        Args:
+            path: The path of the object to be persisted
+            object: The object to be persisted.
+        """
+        client = self._object_session.client("s3")
+
+        client.put_object(
+            Bucket=self._bucket_name,
+            Key=path.get_path(strategy=PathStrategy.Hierarchical),
+            Body=json.dumps(object),
+            ContentType="application/json",
+        )
+
+
+@define
+class LocalFileSystemPersistenceHandler:
+    """Class for persisting JSON serializable dicts locally."""
+
+    folder_path_prefix: str = field(validator=instance_of(str))
+    """The prefix of the folder path where the results are stored."""
+
+    @folder_path_prefix.default
+    def _default_folder_path_prefix(self) -> str:
+        return "./"
+
+    def write_object(self, path: PathConstructor, object: dict) -> None:
+        """Store a JSON serializable dict according to the path.
+
+        This method will store an JSON serializable dict in a local file system.
+
+        Args:
+            path: The path of the object to be persisted
+            object: The object to be persisted.
+        """
+        path_object = Path(
+            f"{self.folder_path_prefix}/{path.get_path(strategy=PathStrategy.Flat)}"
```

Comment:
If I remember correctly (please double check!) you need to use the `/` as separator between different `str` parts of the `Path` and not within the `str`, i.e., have 
`Path(self.folder_path_prefix / path.get_path(strategy=PathStrategy.Flat))`

Reply:
LGTM